### PR TITLE
Extract AddArgument modules

### PR DESCRIPTION
**Pull request by santiweight**
Pure refactor which starts the end of the oversized modules in hls-refactor-plugin.

@pepeiborra and @michaelpj both raised this during review.

Reviewing commit-by-commit will be easier for you

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/3339"></a>

### CODE REVIEW

File: plugins/hls-refactor-plugin/test/Main.hs [hunk 1]
```diff
@@ -2169,220 +2169,21 @@ insertNewDefinitionTests = testGroup "insert new definition actions"
 addFunctionArgumentTests :: TestTree
 addFunctionArgumentTests =
   testGroup
-    "add function argument"
-    [ testSession "simple" $ do
```
Comment (michaelpj) [inline plugins/hls-refactor-plugin/test/Main.hs:2173]: Some of these tests seem to have vanished? Is that deliberate/are the new tests equally good?

**Review by michaelpj — APPROVED** [review]
Looks great, thanks!

**Comment by michaelpj** [comment]
Looks like it has some actual compile errors.

**Comment by santiweight** [comment]
Now builds on 9.2.4 and 8.10.7. I'll let CI tell me about the other versions...

**Comment by santiweight** [comment]
There was some flakey test in call-hierarchy on Ubuntu; I reran and it worked. So I'm going to try to merge now...

**Comment by santiweight** [comment]
Looks like an erroneous failure to me?

**Pull request merged**